File: data_structure/1_linked_list/1_find_intersection/find_intersection.py
def getIntersectionNode( headA,headB):
    l1 = headA
    l2 = headB
    while l1 != l2:
        if not l1: l1 = headB
        else: l1 = l1.next

        if not l2: l2 = headA
        else: l2 = l2.next

    return l1

File: data_structure/1_linked_list/1_find_intersection/find_intersection.js
'use strict';

function find_intersection (headA, headB){

  let l1 = headA, l2 = headB;

  while(l1 !== l2){
    l1 = (l1 === null) ? headB:l1.next;
    l2 = (l2 === null) ? headA:l2.next;
  }


  return l1;
}

module.exports=find_intersection;
